common/persistence/Pinot/pinotVisibilityStore.go

```diff
@@ -471,28 +517,36 @@ func (v *pinotVisibilityStore) CountWorkflowExecutions(ctx context.Context, requ
 	}, nil
 }
 
-func (v *pinotVisibilityStore) DeleteWorkflowExecution(
-	ctx context.Context,
-	request *p.VisibilityDeleteWorkflowExecutionRequest,
-) error {
-	return &types.BadRequestError{Message: "Operation is not supported. Pinot doesn't support this operation so far."}
-}
-
-func (v *pinotVisibilityStore) DeleteUninitializedWorkflowExecution(
-	ctx context.Context,
-	request *p.VisibilityDeleteWorkflowExecutionRequest,
-) error {
-	// temporary: not implemented, only implemented for ES
-	return &types.BadRequestError{Message: "Operation is not supported. Pinot doesn't support this operation so far."}
-}
-
 func (v *pinotVisibilityStore) checkProducer() {
 	if v.producer == nil {
 		// must be bug, check history setup
 		panic("message producer is nil")
 	}
 }
 
+func createDeleteVisibilityMessage(domainID string,
```

Comment:
I'm think if we can combine with createVisibilityMessage using if isDeleted to return different message. But for now, at least we need a comment here to say why we have new method here and just have these fields since it is too similar to the createVisibilityMessage method so other people can understand

Reply:
comment added